### ISSUE TITLE
Use <pre> when displaying plain-text project descriptions

### ISSUE THREAD

**Issue by ishirav** [pypi/warehouse#4008]
**What's the problem this feature will solve?**
When packages provide a plain-text description, they often use indentation to format the text, or sometimes ASCII-art tables and such. Rendering these without preserving whitespace, and without using a monospaced font, breaks the formatting.

This is relevant also for rst descriptions that could not be converted to HTML (due to errors), and are displayed as plain-text instead.

**Describe the solution you'd like**
Preserve whitespace and use a monospaced font when displaying plain-text descriptions. This can be achieved by wrapping them in a `<pre>` tag, or by applying CSS along the lines of:
```
white-space: pre;
font-family: monospace;
```

**Additional context**
Examples of affected packages:
- https://pypi.org/project/PenSec/#description
- https://pypi.org/project/picalo/#description

**Comment by nlhkabu** [comment]
@di is there a way to differentiate between plain text / markdown / rst descriptions?  If so, can we add this information in a css class so we can target a style to plain text only?  e.g. 

* `vertical-tabs__content vertical-tabs__content--plain-text`
* `vertical-tabs__content vertical-tabs__content--markdown`
* `vertical-tabs__content vertical-tabs__content--rst`

**Comment by ewdurbin** [comment]
Here's a basic diff that would allow us to return the renderer that was ultimately used along with the rendered description:

```diff
diff --git a/tests/unit/utils/test_readme.py b/tests/unit/utils/test_readme.py
index 6b646498..3449c61e 100644
--- a/tests/unit/utils/test_readme.py
+++ b/tests/unit/utils/test_readme.py
@@ -14,32 +14,37 @@ from warehouse.utils import readme
 
 
 def test_render_with_none():
-    result = readme.render(None)
+    renderer, result = readme.render(None)
     assert result is None
 
 
 def test_can_render_rst():
-    result = readme.render("raw thing", "text/x-rst")
+    renderer, result = readme.render("raw thing", "text/x-rst")
+    assert renderer == 'rst'
     assert result == "<p>raw thing</p>\n"
 
 
 def test_cant_render_rst():
-    result = readme.render("raw `<thing", "text/x-rst")
+    renderer, result = readme.render("raw `<thing", "text/x-rst")
+    assert renderer == 'plain'
     assert result == "raw `&lt;thing"
 
 
 def test_can_render_plaintext():
-    result = readme.render("raw thing", "text/plain")
+    renderer, result = readme.render("raw thing", "text/plain")
+    assert renderer == 'plain'
     assert result == "raw thing"
 
 
 def test_can_render_markdown():
-    result = readme.render("raw thing", "text/markdown")
+    renderer, result = readme.render("raw thing", "text/markdown")
+    assert renderer == 'markdown'
     assert result == "<p>raw thing</p>\n"
 
 
 def test_can_render_missing_content_type():
-    result = readme.render("raw thing")
+    renderer, result = readme.render("raw thing")
+    assert renderer == 'rst'
     assert result == "<p>raw thing</p>\n"
 
 
diff --git a/warehouse/utils/readme.py b/warehouse/utils/readme.py
index 5b2df300..0f9bff32 100644
--- a/warehouse/utils/readme.py
+++ b/warehouse/utils/readme.py
@@ -21,22 +21,22 @@ import readme_renderer.txt
 
 
 _RENDERERS = {
-    None: readme_renderer.rst,  # Default if description_content_type is None
-    "": readme_renderer.rst,  # Default if description_content_type is None
-    "text/plain": readme_renderer.txt,
-    "text/x-rst": readme_renderer.rst,
-    "text/markdown": readme_renderer.markdown,
+    None: ('rst', readme_renderer.rst),  # Default if description_content_type is None
+    "": ('rst', readme_renderer.rst),  # Default if description_content_type is None
+    "text/plain": ('plain', readme_renderer.txt),
+    "text/x-rst": ('rst', readme_renderer.rst),
+    "text/markdown": ('markdown', readme_renderer.markdown),
 }
 
 
 def render(value, content_type=None, use_fallback=True):
     if value is None:
-        return value
+        return None, value
 
     content_type, parameters = cgi.parse_header(content_type or "")
 
     # Get the appropriate renderer
-    renderer = _RENDERERS.get(content_type, readme_renderer.txt)
+    renderer_type, renderer = _RENDERERS.get(content_type, readme_renderer.txt)
 
     # Actually render the given value, this will not only render the value, but
     # also ensure that it's had any disallowed markup removed.
@@ -47,9 +47,10 @@ def render(value, content_type=None, use_fallback=True):
     # that readme_renderer will deal with sanitizing the content.
     # Skip the fallback option when validating that rendered output is ok.
     if use_fallback and rendered is None:
+        renderer_type = "plain"
         rendered = readme_renderer.txt.render(value)
 
-    return rendered
+    return renderer_type, rendered
 
 
 def renderer_version():
```

From there, we'd need to update the `release_detail` view to accept the new return

https://github.com/pypa/warehouse/blob/9738edef42bd8c43656ef85a4c3ddaf5ed534002/warehouse/packaging/views.py#L85

and pass that on into the template for rendering

https://github.com/pypa/warehouse/blob/9738edef42bd8c43656ef85a4c3ddaf5ed534002/warehouse/packaging/views.py#L117

Then add the necessary bits to the template

https://github.com/pypa/warehouse/blob/9738edef42bd8c43656ef85a4c3ddaf5ed534002/warehouse/templates/packaging/detail.html#L161-L173

From there it's just CSS? I'm not sure :)

**Comment by di** [comment]
Fixed in #12408.
